cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_dense_default_batch_container.h

```diff
@@ -53,7 +54,7 @@ BatchContainer<algorithmFpType, method, cpu>::~BatchContainer()
 template <typename algorithmFpType, Method method, CpuType cpu>
 services::Status BatchContainer<algorithmFpType, method, cpu>::compute()
 {
-    const daal::algorithms::Parameter * const par            = _par;
+    const Parameter * const par                              = static_cast<const Parameter *>(_par);
```

Comment:
Why you need this cast?

Reply:
_par is a pointer to base class. I need a pointer to derived class to access par->resultsToEvaluate